src/coreclr/jit/compiler.h

```diff
@@ -10011,13 +10014,22 @@ class Compiler
         }
 #endif // !DEBUG
 
+        // TODO: we should convert these into a single OptimizationLevel
+
         bool OptimizationDisabled() const
         {
-            return MinOpts() || compDbgCode;
+            assert(compMinOptsIsSet);
+            return !canUseAllOpts;
         }
         bool OptimizationEnabled() const
         {
-            return !OptimizationDisabled();
+            assert(compMinOptsIsSet);
+            return canUseAllOpts;
+        }
+        bool Tier0OptimizationEnabled() const
```

Comment:
A minor nit is that `OptimizationEnabled` and `Tier0OptimizationEnabled` both existing may be a bit confusing. Is there any idea what we would call the levels if this were some kind of `OptimizationLevel` enum, just for ideas on what we could call it as an alternative?

I think this is fine for now, because getting better names is a more involved change; just wanted to check if there were any other ideas/suggestions for the interim

Reply:
Yep, I agree that it doesn't look pretty, but a proper refactoring will be quite big and hard to review

Reply:
> any idea what we would call the levels if this were some kind of OptimizationLevel enum, just for ideas on what we could call it as an alternative?

Not yet. Andy suggested to use OptLevel: NoOpts, Size, Speed where Size implies "light-weight". The problem that we already have optLevel in jit, but we never use it (it's always Blended) so it all needs some re-thinking